app/common/elasticsearchclient.py

```diff
@@ -240,19 +240,25 @@ def best_hit_search(self,
         # current_app.logger.debug("Got %d Hits in %ims" % (res['hits']['total'], res['took']))
         data = []
         
-        for res in results['responses']:
-            res_data = [];
-            for hit in res['hits']['hits']:
+        #there are len(params.q) responses - one per query
+        for i in range(len(results['responses'])):
+            name = params.q[i]  #even though we are guaranteed that responses come back in order, and can match query to the result - this might be convenient to have       
+            res = results['responses'][i]
+            if(len(res['hits']['hits'])>0):
```

Comment:
it is safer to use `if res['hits']['total']:` there might not be a `res['hits']['hits']`